### PR TITLE
Add magnetic dipole and nabla operator

### PR DESCRIPTION
**Pull request by maxscheurer**
This PR adds dumping of the magnetic dipole and nabla operator from `pyscf`.

### CODE REVIEW

File: adcctestdata/dump_pyscf.py [hunk 1]
```diff
@@ -217,5 +217,16 @@ def dump_pyscf(scfres, out):
     mmp.create_dataset("elec_1",
                        data=scfres.mol.intor_symmetric("int1e_r", comp=3))
 
+    varop = data.create_group("uncategorized_operators")
```
Comment (mfherbst) [inline adcctestdata/dump_pyscf.py]: British spelling, please: "uncategorised_operators"

Also: Better introduce groups "magnetic_moments" and "derivatives"? The problem is that changing the name of the group is a lot of hassle (it occurs in the DataHfProvider in adcc, the test setup, ...), , so we should choose the name wisely and "uncategorised_operators" opens the door to a lot of quick and dirty dumps of operators in the future, which I'm not sure is the wisest way to go.

Comment (maxscheurer) [reply adcctestdata/dump_pyscf.py]: Oh come on British spelling...
I'd go for `derivativezzz` 😉

Comment (mfherbst) [reply adcctestdata/dump_pyscf.py]: That's what's used basically everywhere in adcc. We can discuss about changing, but then you do the renaming ;).

Comment (maxscheurer) [reply adcctestdata/dump_pyscf.py]: Done.

Comment (maxscheurer) [reply adcctestdata/dump_pyscf.py]: I was just kidding... 😉

File: adcctestdata/dump_pyscf.py [hunk 2]
```diff
@@ -217,5 +217,17 @@ def dump_pyscf(scfres, out):
     mmp.create_dataset("elec_1",
                        data=scfres.mol.intor_symmetric("int1e_r", comp=3))
 
+    magm = data.create_group("magnetic_moments")
+    derivs = data.create_group("derivatives")
+    with scfres.mol.with_common_orig([0.0, 0.0, 0.0]):
+        magm.create_dataset(
+            "mag_1",
+            data=0.5 * scfres.mol.intor('int1e_cg_irxp', comp=3, hermi=2)
+        )
```
Comment (mfherbst) [inline adcctestdata/dump_pyscf.py:226]: Can you format as in 217 to 218 ... I think in this case I would vote for brevity.

**Review by mfherbst — APPROVED** [review]
LGTM.

File: adcctestdata/dump_pyscf.py (hunk 2 above)
Comment (mfherbst) [reply adcctestdata/dump_pyscf.py:226]: also in the part below of course ;)

Comment (maxscheurer) [reply adcctestdata/dump_pyscf.py:226]: I cannot because `line too long`.

Comment (mfherbst) [reply adcctestdata/dump_pyscf.py:226]: Cool ;)

**Comment by mfherbst** [comment]
You probably need a release for pinning in adcc I guess?

**Comment by maxscheurer** [comment]
I mean I can also just pin the current master? However you like 😄

**Comment by mfherbst** [comment]
Done.

**Pull request merged**